### DISCUSSION TITLE
LANraragi v.0.7.9 - Lucy Can't Dance

### DISCUSSION THREAD

**Discussion by Difegue** [Difegue/LANraragi#483] [Show and tell]
![0e7ea1959c878b8b997942f001a92a7e61f348c242613d280d30d0313ac1951b](https://user-images.githubusercontent.com/8237712/121266905-9f2e8e00-c8bb-11eb-8282-e0a82ea7810a.jpg)  

🥵 Hot hot release comin' thru!   
Despite the fact I was away working on not-LRR things, I received a surprising amount of external contributions which make up the majority of the changes here.  

I'd like to once again thank @nonamethanks and @koma-cute for the work they've put in on this version! 🤠  

## The Web Reader has been entirely overhauled ✨( + #242 !)

![image](https://user-images.githubusercontent.com/8237712/121267231-2e3ba600-c8bc-11eb-8efd-1f8696f74cf4.png)

Following up on the UI improvements from 0.7.8, the guts of the reader have been rewritten to be more lean, have **way** more keyboard shortcuts, and be easier for userscripts to hook into.  

Speaking of userscripts, while this revamp breaks the venerable infinite scrolling userscript, it also finally brings said infinite scrolling as a built-in option! AIn't that great. 

## Server-Side Progression tracking is now optional 👀 (#474)

![](https://user-images.githubusercontent.com/8237712/120354004-69f5cf00-c302-11eb-88cf-893be821d579.png)

While moving reading progression tracking to the server makes single-user installs way more powerful, if you host LRR on a remote server to uhhhhhhhhhhhhh `share the good stuff` with friends, things would get weird fast as users would constantly overwrite each other's progress.

I don't fancy adding a full account system, as it doesn't feel very worthwhile to do so, (Public LRR instances are quite rare...) but what I can easily do is bring back the former clientside progress tracking with `localStorage` as a server-side preference. 🙌

## A **lot** of Download System Bugfixes/Improvements

![c15302f7a6528d28648b28fd87923f19270c4058c7473180ee94f122dad111a9](https://user-images.githubusercontent.com/8237712/121268875-16195600-c8bf-11eb-93b4-c0c13315b35d.png)

It really do be like that sometimes. 

* ### General
  🌮 (#449) Prevent downloaded filenames from containing illegal characters and exceeding 255 characters (thanks Windows I guess)
  🍔 (#464) Fix Downloader errors not bubbling up properly to the end-user
  🍟 Retry URL downloading attempts up to 5 times if they fail due to a network issue
* ### Fakku
  🌭 (#476) Add an explicit error if you get a CloudFlare block on Fakku 
  🥓 (#476) Add "Direction" to the columns that we don't parse to avoid errors on select galleries
* ### E-H
  🧇 (#465) Fix autostart behavior sometimes not giving a correct URL to download
* ### NH
  🥞 (#473) NHentai plugin can now try to find and use gallery IDs from archive file names

The [Tsukihi](https://github.com/Difegue/Tsukihi/releases/tag/v.0.9) browser extension also received a minor update last month, if you didn't catch it on Discord.

## Other Changes

* ### API Changes
  🔑 `/api/info` now exposes whether the server has serverside progression tracking enabled or not. That's all!    

![](https://user-images.githubusercontent.com/8237712/120355114-50a15280-c303-11eb-8ec1-ccc216ff01d8.png)

* ### Miscellaneous nonsense
  💎 Reader: Fix Spacebar keybind incorrectly going backwards on right-to-left mode
  💎 Add eslint configuration to the GitHub codespaces container, matching the eslint standards now enforced on the JS side of the app
  💎 Prevent fetching all archives when category search returns no results
  💎 Don't add "other" namespaces when searching for an unnamespaced tag
  💎 Move `shinobu.pid` and `minion.pid` to the app temporary folder instead of `script`
  💎 (#469) Use Prefork instead of Hypnotoad as the production server to easily move its pid_file, and move said file to the temp folder as well
  💎 (#447) Reorder Context Menu items to show "Read" and "Edit" first even if logged in
  💎 (#478) Remove very old UTF-8 encoding workaround in title/MOTD 
  💎 (#466) Clear out tooltips after a search 
  💎 Don't remove the temporary folder itself when running `clean_temp_full` to avoid an error message on Docker installs 
  💎 CSS touchups to fix some alignment issues in the Configuration page
# 

_Bowie’s general dismissal of Madonna is understandable, if petty and regrettable (imagine the music the two could have made together).
So “Lucy Can’t Dance” is a piece of well-aimed snark, targeting Madonna at a time when her cultural presence was inescapable (it was the era of the [Sex](http://en.wikipedia.org/wiki/Sex_%28book%29) book, one of the most tedious and expensive pieces of pornography ever released)._

<hr /><em>This discussion was created from the release <a href='https://github.com/Difegue/LANraragi/releases/tag/v.0.7.9'>LANraragi v.0.7.9 - Lucy Can't Dance</a>.</em>

**Comment by DevilGlitch** [comment]
hmmmm
